logemongo

```diff
@@ -30,23 +35,83 @@ GetOptions(\%OPTIONS, qw/
 	password|pass|p=s
 	include|i=s@
 	exclude|x=s@
+	filters|filter=s%
 	debug|D!
 	count|c=i
+	detail!
 /) or pod2usage(2);
 pod2usage(1) if $OPTIONS{help};
 
+# compile all the regexen for SPEED!
+my @INCLUDES = map { qr/$_/ } @{$OPTIONS{include}};
+my @EXCLUDES = map { qr/$_/ } @{$OPTIONS{exclude}};
+
 sub DEBUG {
 	$OPTIONS{debug} ? print "[DEBUG] @_\n" : print "";
 	return;
 }
 
-# compile all the regexen for SPEED!
-my @INCLUDES = map { qr/$_/ } @{$OPTIONS{include}};
-my @EXCLUDES = map { qr/$_/ } @{$OPTIONS{exclude}};
+sub handle {
+	my ($message) = @_;
+	my ($logdata, $payload, %msghash);
+	if ($message =~ m/(\[NXLOG@\d+(.*)\]) (.*)/) {
+		$logdata = $2;
+		$payload = $3;
+	} else {
+		return 0;
+	}
+	($msghash{'timestamp'}, $msghash{'uuid'}) =  split ' ', $message;
+	my $qr = qr/\s*([^=,\s]+)\s*=\s*(?:"((?:[^"]|"")*)"|([^,"]*))\s*,?/;
+	while ($logdata =~ m/$qr/) {
+		$msghash{$1}=$2;
+		$logdata =~ s/$qr//;
+	}
```

Comment:
After a substitution, the `$1` and `$2` back refs will still be useful.  You don't need the `qr//` here.

Consider:

```
my $log = "key=value other=value";
while ($log =~ s/(\w*?)=([^ ]*)//) {
    print "found '$1' = '$2'\n";
}
```